### PR TITLE
Add pickle jar with cooldown logic

### PR DESCRIPTION
**Pull request by yiepier**
- Add pickle jar with cooldown logic

### CODE REVIEW

File: src/tests/strategies/polygon/balancer/testBalancerBase.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+      const expectedDevFund = _after.mul(50).mul(4).div(1000).div(7);
+
+      const _devAfter = await want.balanceOf(devfund.address);
+      _treasuryAfter = await want.balanceOf(treasury.address);
+      console.log("\n👨‍🌾 Dev balance after picklejar withdrawal: ", _devAfter.toString());
+      console.log("💸 Treasury balance after picklejar withdrawal: ", _treasuryAfter.toString());
+
+      //0% goes to treasury
+      const _treasuryFund = _treasuryAfter.sub(_treasuryBefore);
+      expect(_treasuryFund).to.be.eq(0, "treasury've stolen money!!!!");
```
Comment (larrythecucumber321) [inline src/tests/strategies/polygon/balancer/testBalancerBase.js:138]: You have a `_devBefore` and a `_devAfter`. I assume you meant to include an assertion that the difference is equal to `expectedDevFund`, correct?

File: src/tests/strategies/polygon/balancer/testBalancerBase.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+      //withdraw
+      const _devBefore = await want.balanceOf(devfund.address);
+      _treasuryBefore = await want.balanceOf(treasury.address);
+      console.log("\n👨‍🌾 Dev balance before picklejar withdrawal: ", _devBefore.toString());
+      console.log("💸 Treasury balance before picklejar withdrawal: ", _treasuryBefore.toString());
+
+      const _cooldown = await pickleJar.balanceOf(alice.address);
+
+      await pickleJar.cooldown(_cooldown);
```
Comment (larrythecucumber321) [inline src/tests/strategies/polygon/balancer/testBalancerBase.js:121]: I see that `cooldown` is set for the Alice somewhere somewhat random in the flow. Is there a particular part when it should be called? Why not bake it into the `deposit` function? See the Adamant contract, which does this.

File: src/pickle-jar-cooldown.sol [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    uint256 public min = 9500;
+    uint256 public constant max = 10000;
+
+    address public governance;
+    address public timelock;
+    address public controller;
+
+    uint256 public cooldownTime = 7 days;
+    uint256 public initialWithdrawalFee = 50;
+    uint256 public initialWithdrawalFeeMax = 1000;
```
[the remark this answers is not in the document]
Comment (mannyreimi88) [reply src/pickle-jar-cooldown.sol:27]: @larrythecucumber321 we calculated the potential gain from trying to time the deposits to imitate a sandwich and came up with 5% as a reasonable penalty. Of course, this varies per strategy.

[the remark this answers is not in the document]
Comment (yiepier) [reply src/pickle-jar-cooldown.sol:27]: So 1000 or 10000, which one should we take finally?

File: src/tests/strategies/polygon/balancer/testBalancerBase.js (hunk 1 above)
Comment (yiepier) [reply src/tests/strategies/polygon/balancer/testBalancerBase.js:132]: correct, updated it

**Pull request merged**